kf_lib_data_ingest/common/file_retriever.py

```diff
@@ -225,10 +287,20 @@ def get(self, url, auth=None):
                     dir=self.storage_dir,
                     delete=self.cleanup_at_exit and not self.__tmpdir
                 )
+
+                # Set auth config if there is one and validate it
+                if self.auth_config:
+                    self._validate_auth_config(self.auth_config)
+
                 # Fetch the remote data
+                kwargs = {
+                    'auth': auth,
+                    'logger': self.logger,
+                    'auth_config': self.auth_config
```

Comment:
I wonder if we should keep `auth_config` specific to http type gets (and call it `http_auth_confg`) since we don't quite know how it will be used for the other types of gets. I could see it being used in the s3 get maybe to get AWS access credentials? Idk..